### PR TITLE
Use ReadOnlySpan<char> to avoid Substring allocation

### PR DESCRIPTION
**Pull request by meziantou**
Use `ReadOnlySpan<char>` instead of `string`, so you can replace `Substring` with `Slice` and remove 2 allocations

### CODE REVIEW

**Review by JamesNK — APPROVED** [review]
LGTM

File: src/Http/WebUtilities/src/QueryHelpers.cs [hunk 1]
```diff
@@ -113,13 +113,13 @@ public static string AddQueryString(string uri, IEnumerable<KeyValuePair<string,
         }
 
         var anchorIndex = uri.IndexOf('#');
-        var uriToBeAppended = uri;
-        var anchorText = "";
+        ReadOnlySpan<char> uriToBeAppended = uri;
+        ReadOnlySpan<char> anchorText = ReadOnlySpan<char>.Empty;
```
Comment (gfoidl) [inline src/Http/WebUtilities/src/QueryHelpers.cs:117]:
```suggestion
        var uriToBeAppended = uri.AsSpan();
        var anchorText = ReadOnlySpan<char>.Empty;
```
To use more `var` here 😉

**Comment by gfoidl** [comment]
There are [more SubString](https://grep.app/search?q=SubString&filter[lang][0]=C%23&filter[repo][0]=dotnet/aspnetcore)s in aspnetcore. Should all of these be audited? (In general, not necessarily in this PR)

**Comment by JamesNK** [comment]
Sure. I wouldn’t be surprised if there is an analyzer for this.

**Comment by gfoidl** [comment]
Ah, exactly: [CA1846](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1846)

**Comment by BrennanConroy** [comment]
CA1846 is already enabled for this repo. But it's obviously a bit limited in what it can find, case and point this PR 😄

**Comment by Tratcher** [comment]
Thanks

**Pull request merged**